jwql/instrument_monitors/miri_monitors/data_trending/plots/temperature_tab.py

```diff
@@ -0,0 +1,348 @@
+    h = pf.add_to_plot(p, "DIG SW", "IMIR_SW_SCE_DIG_TEMP", start, end, conn, color = "cyan")
+    i = pf.add_to_plot(p, "DIG LW", "IMIR_LW_SCE_DIG_TEMP", start, end, conn, color = "darkblue")
+
+    pf.add_hover_tool(p,[a,b,c,d,e,f,g,h,i,j,k])
+
+    p.legend.location = "bottom_right"
+    p.legend.orientation = "horizontal"
+    p.legend.click_policy = "hide"
+
+    return p
+
+def det(conn, start, end):
+    '''Create specific plot and return plot object
+    Parameters
+    ----------
+    conn : DBobject
+        Connection object that represents database
+    start : time
+        Startlimit for x-axis and query (typ. datetime.now()- 4Months)
+    end : time
+        Endlimit for x-axis and query (typ. datetime.now())
+    Return
+    ------
+    p : Plot object
+        Bokeh plot
+    '''
+
+    # create a new plot with a title and axis labels
+    p = figure( tools = "pan,wheel_zoom,box_zoom,reset,save",       \
+                toolbar_location = "above",                         \
+                plot_width = 1120,                                   \
+                plot_height = 400,                                  \
+                y_range = [6.395,6.41],                             \
+                x_axis_type = 'datetime',                           \
+                output_backend="webgl",                             \
+                x_axis_label = 'Date', y_axis_label = 'Temperature (K)')
+
+    p.grid.visible = True
+    p.title.text = "Detector Temperature"
+    pf.add_basic_layout(p)
+
+    a = pf.add_to_plot(p, "Det. Temp. IC", "IGDP_MIR_IC_DET_TEMP", start, end, conn, color = "red")
+    b = pf.add_to_plot(p, "Det. Temp. LW", "IGDP_MIR_LW_DET_TEMP", start, end, conn, color = "green")
+    c = pf.add_to_plot(p, "Det. Temp. SW", "IGDP_MIR_SW_DET_TEMP", start, end, conn, color = "blue")
+
+    pf.add_hover_tool(p,[a,b,c])
+
+    p.legend.location = "bottom_right"
+    p.legend.orientation = "horizontal"
+    p.legend.click_policy = "hide"
+
+    return p
+
+def temperature_plots(conn, start, end):
+    '''Combines plots to a tab
+    Parameters
+    ----------
+    conn : DBobject
+        Connection object that represents database
+    start : time
+        Startlimit for x-axis and query (typ. datetime.now()- 4Months)
+    end : time
+        Endlimit for x-axis and query (typ. datetime.now())
+    Return
+    ------
+    p : tab object
+        used by dashboard.py to set up dashboard
+    '''
+
+    descr = Div(text=
+    """
+    <style>
+    table, th, td {
+      border: 1px solid black;
+      background-color: #efefef;
+      border-collapse: collapse;
+      padding: 5px
+    }
+    </style>
+
+    <body>
+    <table style="width:100%">
+      <tr>
+        <th><h6>Plotname</h6></th>
+        <th><h6>Mnemonic</h6></th>
+        <th><h6>Description</h6></th>
+      </tr>
+      <tr>
+        <td>CRYO Temperatures</td>
+        <td>IGDP_MIR_ICE_T1P_CRYO<br>
```

Comment:
I would suggest to store this type of information in configurations files, which will improve maintainability.
Can some of the information be pulled from the EDB mnemonic description (despite it being incomplete)?

Reply:
Sounds very good! Would you do one configuration file for all tabs or one each? Would you recommend implementing it in html or python?

Reply:
I would suggest simple text files, possibly comma-separated tables/lists, that are read by the code.